napari/layers/image/image.py

```diff
@@ -184,11 +200,18 @@ def __init__(
         self.is_pyramid = is_pyramid
         self.rgb = rgb
         self._data = data
-        self._data_pyramid = data_pyramid
         if self.is_pyramid:
-            self._data_level = len(data_pyramid) - 1
+            self._data_level = len(self.data) - 1
+            big_levels = [
+                np.any(np.greater_equal(p.shape, 2 ** 6 - 1)) for p in data
+            ]
+            if np.any(big_levels):
+                self._thumbnail_level = np.where(big_levels)[0][-1]
+            else:
+                self._thumbnail_level = 0
```

Comment:
can you explain this in a comment? I don't understand what this code is doing. What's a big level?

Reply:
comment made this is to find a good level to use for thumbnail generation - not too small so that you just say have a 2x2 pixel image for the thumbnail (which might be the top of the pyramid)

Reply:
I might call that a goldilocks level. ;)